x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/AbstractInferenceServiceTests.java

```diff
@@ -51,132 +39,64 @@
  * </p>
  */
-public abstract class AbstractInferenceServiceTests extends InferenceServiceTestCase {
-
-    protected final MockWebServer webServer = new MockWebServer();
-    protected ThreadPool threadPool;
-    protected HttpClientManager clientManager;
-
-    @Override
-    @Before
-    public void setUp() throws Exception {
-        super.setUp();
-        webServer.start();
-        threadPool = createThreadPool(inferenceUtilityExecutors());
-        clientManager = HttpClientManager.create(Settings.EMPTY, threadPool, mockClusterServiceEmpty(), mock(ThrottlerManager.class));
-    }
-
-    @Override
-    @After
-    public void tearDown() throws Exception {
-        super.tearDown();
-        clientManager.close();
-        terminate(threadPool);
-        webServer.close();
-    }
-
-    private final TestConfiguration testConfiguration;
+public abstract class AbstractInferenceServiceTests extends AbstractInferenceServiceBaseTests {
 
     public AbstractInferenceServiceTests(TestConfiguration testConfiguration) {
-        this.testConfiguration = Objects.requireNonNull(testConfiguration);
-    }
-
-    /**
-     * Main configurations for the tests
-     */
-    public record TestConfiguration(CommonConfig commonConfig, UpdateModelConfiguration updateModelConfiguration) {
-        public static class Builder {
-            private final CommonConfig commonConfig;
-            private UpdateModelConfiguration updateModelConfiguration = DISABLED_UPDATE_MODEL_TESTS;
-
-            public Builder(CommonConfig commonConfig) {
-                this.commonConfig = commonConfig;
-            }
-
-            public Builder enableUpdateModelTests(UpdateModelConfiguration updateModelConfiguration) {
-                this.updateModelConfiguration = updateModelConfiguration;
-                return this;
-            }
-
-            public TestConfiguration build() {
-                return new TestConfiguration(commonConfig, updateModelConfiguration);
-            }
-        }
+        super(testConfiguration);
     }
 
-    /**
-     * Configurations that useful for most tests
-     */
-    public abstract static class CommonConfig {
-
-        private final TaskType taskType;
-        private final TaskType unsupportedTaskType;
-
-        public CommonConfig(TaskType taskType, @Nullable TaskType unsupportedTaskType) {
-            this.taskType = Objects.requireNonNull(taskType);
-            this.unsupportedTaskType = unsupportedTaskType;
-        }
-
-        protected abstract SenderService createService(ThreadPool threadPool, HttpClientManager clientManager);
-
-        protected abstract Map<String, Object> createServiceSettingsMap(TaskType taskType);
-
-        protected abstract Map<String, Object> createTaskSettingsMap();
-
-        protected abstract Map<String, Object> createSecretSettingsMap();
+    public void testParseRequestConfig_CreatesAnEmbeddingsModel() throws Exception {
```

Comment:
The `*ParseRequestConfig_*` tests could be converted into parameterized tests as well. I decided to not do it in this PR because it's already quite a few changes.